### PR TITLE
fix: react-native-screens not building on visionOS

### PR DESCRIPTION
**Pull request by okwasniewski**
## Description

This PR fixes React Native screens not building on visionOS. As this platform needs explicit imports for used modules

## Changes

- Fixed building `RNSConvert.h` for visionOS

## Test code and steps to reproduce

Compile on visionOS

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Updated documentation: 
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/guides/GUIDE_FOR_LIBRARY_AUTHORS.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/native-stack/README.md
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/types.tsx
  - [ ] https://github.com/software-mansion/react-native-screens/blob/main/src/native-stack/types.tsx
- [ ] Ensured that CI passes

### CODE REVIEW

**Comment by okwasniewski** [comment]
Hey! What do you think about implementing visionOS CI to ensure new changes don't break existing platform support?

**Comment by tboba** [comment]
Hey @okwasniewski! Yeah, I think we could add a new action for that 👍

**Comment by tboba** [comment]
@okwasniewski hmm, could you send some repository with the example you've migrated to react-native-test-app? I've got some concerns about migrating to it, but before asking them I need to check if in my head these questions are valid 😄

**Comment by okwasniewski** [comment]
@tboba There you go: https://github.com/callstack/repack/pull/629

**Comment by okwasniewski** [comment]
@tboba I think adding visionOS CI can be done in a separate PR, do you think we can merge this one?

**Comment by kkafar** [comment]
I'll get in a way here a bit. Can we get a link to documentation where it states that explicit imports are required? How is this enforced? I can see it only being done via compiler option & I'm not aware of such one (I obviously don't know vast majority of them, but quick googling did not help). 

I'm asking these questions cause there is possibility that the issue lies somewhere else.

**Comment by okwasniewski** [comment]
@kkafar It's missing references for `UISemanticContentAttributeForceRightToLeft` / `UISemanticContentAttributeForceLeftToRight` which come from UIKit. 

Technically you should always import the framework you are using but I guess for iOS it's linked anyway. 

Here is a screenshot:

CleanShot 2024-06-26 at 12 33 12@2x

```

In file included from /Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:1:
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.h:13:4: error: expected a type
   13 | + (UISemanticContentAttribute)UISemanticContentAttributeFromCppEquivalent:
      |    ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.h:16:4: error: expected a type
   16 | + (UINavigationItemBackButtonDisplayMode)UINavigationItemBackButtonDisplayModeFromCppEquivalent:
      |    ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:6:4: error: expected a type
    6 | + (UISemanticContentAttribute)UISemanticContentAttributeFromCppEquivalent:
      |    ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:17:4: error: expected a type
   17 | + (UINavigationItemBackButtonDisplayMode)UINavigationItemBackButtonDisplayModeFromCppEquivalent:
      |    ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:11:14: error: use of undeclared identifier 'UISemanticContentAttributeForceRightToLeft'
   11 |       return UISemanticContentAttributeForceRightToLeft;
      |              ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:13:14: error: use of undeclared identifier 'UISemanticContentAttributeForceLeftToRight'
   13 |       return UISemanticContentAttributeForceLeftToRight;
      |              ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:22:14: error: use of undeclared identifier 'UINavigationItemBackButtonDisplayModeDefault'
   22 |       return UINavigationItemBackButtonDisplayModeDefault;
      |              ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:24:14: error: use of undeclared identifier 'UINavigationItemBackButtonDisplayModeGeneric'
   24 |       return UINavigationItemBackButtonDisplayModeGeneric;
      |              ^
/Users/okwasniewski/workspace/SpatialAmbientSounds/node_modules/react-native-screens/ios/RNSConvert.mm:26:14: error: use of undeclared identifier 'UINavigationItemBackButtonDisplayModeMinimal'
   26 |       return UINavigationItemBackButtonDisplayModeMinimal;
      |              ^
5 warnings and 9 errors generated.

```

There is nothing in the docs about the compiler being more strict when compiling for visionOS but that's the thing I've observed when working with this platform.

File: ios/RNSConvert.h [hunk 1]
```diff
@@ -1,6 +1,7 @@
 #ifdef RCT_NEW_ARCH_ENABLED
 #import <react/renderer/components/rnscreens/Props.h>
 #import "RNSEnums.h"
+#import <UIKit/UIKit.h>
```
Comment (kkafar) [inline ios/RNSConvert.h:4]:
```suggestion
#import <UIKit/UIKit.h>
#import <react/renderer/components/rnscreens/Props.h>
#import "RNSEnums.h"
```

Requested change

**Review by kkafar — APPROVED** [review]
Yeah, I took a look now into `RNSConvert.h` and I'm surprised it works right now on iOS, as we do not import any UIKit related headers. I'm not aware of XCode injecting any symbols into translation units, but maybe it does 🤷🏻‍♂️ 

I need to lookup some documentation to figure out why this is the case 😄 

I'll allow myself a liberty of reorganising the imports and I think we're good to go. 


Thanks for this PR!

**Pull request merged**